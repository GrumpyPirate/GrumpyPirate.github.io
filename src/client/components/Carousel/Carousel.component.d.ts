import { ChildrenProps } from 'types';

interface CarouselProps extends ChildrenProps {
  showStatus: boolean;
  showThumbs: boolean;
  infiniteLoop: boolean;
  useKeyboardArrows: boolean;
  transitionTime: number;
  dynamicHeight: boolean;
  showArrows: false;
  emulateTouch: boolean;
  swipeScrollTolerance: number;
}

export { CarouselProps };
