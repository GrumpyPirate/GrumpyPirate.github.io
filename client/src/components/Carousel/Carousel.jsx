import React from 'react';
import PropTypes from 'prop-types';

import './Carousel.scss';

import { Carousel as ReactCarousel } from 'react-responsive-carousel';

const Carousel = ({ children, ...rest }) => <ReactCarousel {...rest}>{children}</ReactCarousel>;

// PropTypes
Carousel.propTypes = {
  children: PropTypes.node.isRequired,
};

export default Carousel;
