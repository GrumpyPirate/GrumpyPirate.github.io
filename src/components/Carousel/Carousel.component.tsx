import React, { FunctionComponent } from 'react';
import { Carousel as ReactCarousel } from 'react-responsive-carousel';

import { CarouselProps } from './Carousel.component.types';

import './Carousel.component.scss';

const Carousel: FunctionComponent<CarouselProps> = ({ children, ...rest }) => (
  <ReactCarousel {...rest}>{children}</ReactCarousel>
);

export default Carousel;
