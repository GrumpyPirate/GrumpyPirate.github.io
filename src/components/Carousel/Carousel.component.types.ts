import { ChildrenProps } from 'types/common';

export interface CarouselProps extends ChildrenProps {
  showStatus: boolean;
  showThumbs: boolean;
  infiniteLoop: boolean;
  useKeyboardArrows: boolean;
  transitionTime: number;
  dynamicHeight: boolean;
  showArrows: false;
  emulateTouch: boolean;
  swipeScrollTolerance: number;
}
