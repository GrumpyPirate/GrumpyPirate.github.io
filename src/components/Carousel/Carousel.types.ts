import { ReactChild } from 'react';

export interface CarouselProps {
  showStatus: boolean;
  showThumbs: boolean;
  infiniteLoop: boolean;
  useKeyboardArrows: boolean;
  transitionTime: number;
  dynamicHeight: boolean;
  showArrows: false;
  emulateTouch: boolean;
  swipeScrollTolerance: number;
  children: ReactChild[];
}
