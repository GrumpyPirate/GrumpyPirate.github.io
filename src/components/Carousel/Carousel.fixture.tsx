import React from 'react';

import { CarouselProps } from './Carousel.types';

const carouselProps: CarouselProps = {
  children: [<div>Test</div>],
  showStatus: false,
  showThumbs: false,
  infiniteLoop: true,
  useKeyboardArrows: true,
  transitionTime: 500,
  dynamicHeight: true,
  showArrows: false,
  emulateTouch: true,
  swipeScrollTolerance: 75,
};

export default carouselProps;
