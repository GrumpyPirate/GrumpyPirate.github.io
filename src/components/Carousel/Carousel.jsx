// React
import React from 'react'
import PropTypes from 'prop-types'

// SCSS
import './Carousel.scss'

// Vendor
import { Carousel as ReactCarousel } from 'react-responsive-carousel'

// Define Carousel
const Carousel = (props) => {
  return (
    <ReactCarousel {...props}>
      {props.children}
    </ReactCarousel>
  )
}

Carousel.propTypes = {
  children: PropTypes.oneOfType(
    [
      PropTypes.arrayOf(PropTypes.node),
      PropTypes.node
    ]
  )
}

export default Carousel
