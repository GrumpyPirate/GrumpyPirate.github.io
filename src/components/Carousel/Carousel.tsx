import React, { FunctionComponent } from 'react';
import { Carousel as ReactCarousel } from 'react-responsive-carousel';

import { CarouselProps } from './Carousel.types';

import './Carousel.scss';

const Carousel: FunctionComponent<CarouselProps> = ({ children, ...rest }) => (
  <ReactCarousel {...rest}>{children}</ReactCarousel>
);

export default Carousel;
