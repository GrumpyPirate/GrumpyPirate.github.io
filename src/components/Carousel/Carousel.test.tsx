import { shallow } from 'enzyme';
import React from 'react';

import Carousel from './Carousel';
import fixture from './Carousel.fixture';

describe('Components', () => {
  describe('Carousel', () => {
    const props = {
      ...fixture,
    };

    it('should render as expected, without crashing', () => {
      const wrapper = shallow(<Carousel {...props} />);

      expect(wrapper).toMatchSnapshot();
    });
  });
});
