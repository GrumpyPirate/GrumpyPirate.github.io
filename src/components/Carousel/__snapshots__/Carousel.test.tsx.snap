// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components Carousel should render as expected, without crashing 1`] = `
<Carousel
  axis="horizontal"
  centerSlidePercentage={80}
  dynamicHeight={true}
  emulateTouch={true}
  infiniteLoop={true}
  interval={3000}
  labels={
    Object {
      "item": "slide item",
      "leftArrow": "previous slide / item",
      "rightArrow": "next slide / item",
    }
  }
  onChange={[Function]}
  onClickItem={[Function]}
  onClickThumb={[Function]}
  onSwipeEnd={[Function]}
  onSwipeMove={[Function]}
  onSwipeStart={[Function]}
  renderArrowNext={[Function]}
  renderArrowPrev={[Function]}
  renderIndicator={[Function]}
  renderItem={[Function]}
  renderThumbs={[Function]}
  selectedItem={0}
  showArrows={false}
  showIndicators={true}
  showStatus={false}
  showThumbs={false}
  statusFormatter={[Function]}
  stopOnHover={true}
  swipeScrollTolerance={75}
  swipeable={true}
  transitionTime={500}
  useKeyboardArrows={true}
  verticalSwipe="standard"
  width="100%"
>
  <div>
    Test
  </div>
</Carousel>
`;
